edb/server/server.py

```diff
@@ -708,11 +724,15 @@ async def _acquire_sys_pgcon(self):
             self._sys_pgcon_waiter.release()
             return None
 
-        if (self.__sys_pgcon is None or
-                not self.__sys_pgcon.is_healthy_to_go_back_to_pool()):
-            self.__sys_pgcon.abort()
-            self.__sys_pgcon = await self._pg_connect(defines.EDGEDB_SYSTEM_DB)
-            await self.__sys_pgcon.set_server(self)
+        if self.__sys_pgcon is None or not self.__sys_pgcon.is_healthy():
+            conn, self.__sys_pgcon = self.__sys_pgcon, None
+            if conn is not None:
+                self._sys_pgcon_ready_evt.clear()
+                conn.abort()
+            # We depend on the reconnect on connection_lost() of __sys_pgcon
+            await self._sys_pgcon_ready_evt.wait()
+            if self.__sys_pgcon is None:
+                self._sys_pgcon_waiter.release()
```

Comment:
Hm, we usually don't check if `_acquire_sys_pgcon` returns an actual connection (not None) before using what it returned. Perhaps we should raise an error if we fail to obtain a connection (and audit all callers of the function?)